### PR TITLE
replace chat_store with a reducer

### PR DESCRIPTION
**Pull request by psinghal20**
Migrate flux store from Mcfly to redux.

### CODE REVIEW

File: app/assets/javascripts/actions/chat_actions.js [hunk 1]
```diff
@@ -1,19 +1,46 @@
...
 
-const ChatActions = Flux.createActions({
-  requestAuthToken() {
-    return API.chatLogin()
-      .then(resp => ({ actionType: 'CHAT_LOGIN_SUCCEEDED', data: resp }))
-      .catch(resp => ({ actionType: 'API_FAIL', data: resp }));
-  },
+export const requestAuthToken = (opts = {}) => dispatch => {
+  return (
+      API.chatLogin()
```
Comment (ragesoss) [inline app/assets/javascripts/actions/chat_actions.js]: Please move these ajax calls out of API, into this file.

**Comment by ragesoss** [comment]
Looks like ChatStore needs to be deleted?

**Comment by psinghal20** [comment]
Yes, it requires some modification forgot to make those. Making them right now. Will ping you when done.

**Comment by psinghal20** [comment]
@mauditecandela I did the suggested changes and shifted the showChat from reducer to the component's state. Please review the changes. Thanks!

File: app/assets/javascripts/utils/api.js [hunk 2]
```diff
@@ -765,38 +765,6 @@ slide_id=${opts.slide_id}`,
...
-          return res(data);
-        }
-      })
-      .fail((obj) => {
-        logErrorMessage(obj);
-        return rej(obj);
-      })
-    );
-  },
-
```
Comment (mauditecandela) [inline app/assets/javascripts/utils/api.js]: nice clean up!

**Comment by psinghal20** [comment]
@ragesoss all the tests seem to pass in the Travis build but there is a failure in language switcher spec. I don't think it is related to this PR. Can you look into this?

**Comment by psinghal20** [comment]
@ragesoss The Travis CI build is failing with failure in language_switcher_spec but I don't think this PR is related to that. Can you please look into it?

**Pull request merged**